src/reason_interface_printer.ml

```diff
@@ -3,18 +3,62 @@ open Ast_404
 
 module Reason_interface_printer : Printer_maker.PRINTER =
     struct
+        type q = Parsetree.signature_item
         type t = Parsetree.signature
-        exception Invalid_config = Printer_maker.Invalid_config
+        let err = Printer_maker.err
 
         (* Note: filename should only be used with .mli files. See reason_toolchain. *)
         let defaultInterfaceParserFor use_stdin filename =
+          let open Reason_toolchain in
+          let (_parser, parsedAsML) =
             if Filename.check_suffix filename ".rei"
-            then (Reason_toolchain.JS.canonical_interface_with_comments (Reason_toolchain.setup_lexbuf use_stdin filename) , false, true)
+            then (JS.canonical_interface_with_comments, false)
             else if Filename.check_suffix filename ".mli"
-            then (Reason_toolchain.ML.canonical_interface_with_comments (Reason_toolchain.setup_lexbuf use_stdin filename), true, true)
-            else (
-                raise (Invalid_config ("Cannot determine default interface parser for filename '" ^ filename ^ "'."))
-                )
+            then (ML.canonical_interface_with_comments, true)
+            else err ("Cannot determine default interface parser for filename '" ^ filename ^ "'.")
+          in
+          _parser (setup_lexbuf use_stdin filename), parsedAsML, true
+
+        let ppx_deriving_runtime =
+          let open Asttypes in
+          let open Parsetree in
+          let open Longident in
+          let open Ast_helper in
+          let open Location in
+          let (@@) f g x = f (g x) in
```

Comment:
@bobzhang I don't see much point in reinventing the wheel if it will end up with roughly the same amount of code. Plus we get the benefit of having it maintained for us.